pandas/io/sas/sas7bdat.py

```diff
@@ -744,7 +744,7 @@ def read(self, nrows: int | None = None) -> DataFrame | None:
             raise EmptyDataError("No columns to parse from file")
 
         if nrows > 0 and self._current_row_in_file_index >= self.row_count:
-            return None
+            return pd.DataFrame()
```

Comment:
Could also keep None but then read_sas will return `DataFrame | None`.

Reply:
I think this breaks `__next__`